compiler/rustc_resolve/src/late.rs

```diff
@@ -1050,10 +1050,75 @@ impl<'a: 'ast, 'ast, 'tcx> Visitor<'ast> for LateResolutionVisitor<'a, '_, 'ast,
         });
         self.diag_metadata.current_function = previous_value;
     }
+
     fn visit_lifetime(&mut self, lifetime: &'ast Lifetime, use_ctxt: visit::LifetimeCtxt) {
         self.resolve_lifetime(lifetime, use_ctxt)
     }
 
+    fn visit_precise_capturing_arg(&mut self, arg: &'ast PreciseCapturingArg) {
+        match arg {
+            // Lower the lifetime regularly; we'll resolve the lifetime and check
+            // it's a parameter later on in HIR lowering.
+            PreciseCapturingArg::Lifetime(_) => visit::walk_precise_capturing_arg(self, arg),
+
+            PreciseCapturingArg::Arg(ident, node_id) => {
```

Comment:
TODO: Document why this is needed.

Specifically, we want `impl use<C>` to try to resolve `C` as both a type parameter *and* a const parameter; whichever one we see first is fine, since we specifically don't allow having two generic params with the same name, even if they're a different namespace.

Reply:
We don't parse this as a list of generic args bc we don't need to support `use<()>`, and we also don't want to require people to capture their consts like `use<{ C }>`.